versioned_docs/version-2.0/LitActions/additionalExamples/generatingSessionKey.md

```diff
@@ -0,0 +1,93 @@
+---
+sidebar_position: 1
+---
+
+# Generating a Session Key
+
+This [example](https://github.com/LIT-Protocol/js-serverless-function-test/blob/main/js-sdkTests/sessionKeys.js) demonstrates how to generate a signed [session key](/SDK/Explanation/authentication/sessionSigs) using Lit Actions.
+
+``` js
+// this example will generate and sign a session key and use it for a request
+
+import * as LitJsSdk from '@lit-protocol/lit-node-client';
+import { Wallet } from "@ethersproject/wallet";
+import { computeAddress } from "@ethersproject/transactions";
+import { SiweMessage } from "siwe";
+
+// this code will be run on the node
+const litActionCode = `
+const go = async () => {
+  // this requests a signature share from the Lit Node
+  // the signature share will be automatically returned in the response from the node
+  // and combined into a full signature by the LitJsSdk for you to use on the client
+  // all the params (toSign, publicKey, sigName) are passed in from the LitJsSdk.executeJs() function
+  const sigShare = await LitActions.signEcdsa({ toSign, publicKey, sigName });
+};
+
+go();
+`;
+
+const runLitAction = async () => {
+  // mock a wallet
+  const wallet = new Wallet(process.env.LIT_MUMBAI_DEPLOYER_PRIVATE_KEY);
+
+  const litNodeClient = new LitJsSdk.LitNodeClient({
+    alertWhenUnauthorized: false,
+    litNetwork: "localhost",
+    debug: true,
+  });
+  await litNodeClient.connect();
+
+  // when the getSessionSigs function is called, it will generate a session key and sign it
+  // using this function. the session key will be used for all requests until the session expires
+  let authNeededCallback = async ({ chain, resources, expiration, uri }) => {
+    const domain = "localhost:3000";
+    const message = new SiweMessage({
+      domain,
+      address: wallet.address,
+      statement: "Sign a session key to use with Lit Protocol",
+      uri,
+      version: "1",
+      chainId: "1",
+      expirationTime: expiration,
+      resources,
+    });
+    const toSign = message.prepareMessage();
+    const signature = await wallet.signMessage(toSign);
+
+    const authSig = {
+      sig: signature,
+      derivedVia: "web3.eth.personal.sign",
+      signedMessage: toSign,
+      address: wallet.address,
+    };
+
+    return authSig;
+  };
+
+  let sessionSigs = await litNodeClient.getSessionSigs({
+    resources: ["litAction://*"],
```

Comment:
Marking this as something we need to change in a future PR - our docs in this section are outdated.